// @ngx-share/core/esm5/lib/share.models.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Share buttons global config
 * @record
 */
export function ShareButtonsConfig() { }
function ShareButtonsConfig_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ShareButtonsConfig.prototype.options;
    /** @type {?|undefined} */
    ShareButtonsConfig.prototype.prop;
}
/**
 * Share buttons global options
 * @record
 */
export function ShareButtonsOptions() { }
function ShareButtonsOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.theme;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.include;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.exclude;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.size;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.url;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.title;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.description;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.image;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.tags;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.twitterAccount;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.autoSetMeta;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.gaTracking;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.windowWidth;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.windowHeight;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.moreButtonIcon;
    /** @type {?|undefined} */
    ShareButtonsOptions.prototype.lessButtonIcon;
}
/**
 * Share buttons collection
 * @record
 */
export function IShareButtons() { }
function IShareButtons_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    IShareButtons.prototype.facebook;
    /** @type {?|undefined} */
    IShareButtons.prototype.twitter;
    /** @type {?|undefined} */
    IShareButtons.prototype.linkedin;
    /** @type {?|undefined} */
    IShareButtons.prototype.google;
    /** @type {?|undefined} */
    IShareButtons.prototype.tumblr;
    /** @type {?|undefined} */
    IShareButtons.prototype.stumble;
    /** @type {?|undefined} */
    IShareButtons.prototype.pinterest;
    /** @type {?|undefined} */
    IShareButtons.prototype.reddit;
    /** @type {?|undefined} */
    IShareButtons.prototype.vk;
    /** @type {?|undefined} */
    IShareButtons.prototype.telegram;
    /** @type {?|undefined} */
    IShareButtons.prototype.messenger;
    /** @type {?|undefined} */
    IShareButtons.prototype.whatsapp;
    /** @type {?|undefined} */
    IShareButtons.prototype.xing;
    /** @type {?|undefined} */
    IShareButtons.prototype.sms;
    /** @type {?|undefined} */
    IShareButtons.prototype.email;
    /** @type {?|undefined} */
    IShareButtons.prototype.print;
    /** @type {?|undefined} */
    IShareButtons.prototype.copy;
}
/**
 * Share button properties
 * @record
 */
export function IShareButton() { }
function IShareButton_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    IShareButton.prototype.text;
    /** @type {?|undefined} */
    IShareButton.prototype.icon;
    /** @type {?|undefined} */
    IShareButton.prototype.type;
    /** @type {?|undefined} */
    IShareButton.prototype.color;
    /** @type {?|undefined} */
    IShareButton.prototype.ariaLabel;
    /** @type {?|undefined} */
    IShareButton.prototype.successText;
    /** @type {?|undefined} */
    IShareButton.prototype.successIcon;
    /** @type {?|undefined} */
    IShareButton.prototype.failText;
    /** @type {?|undefined} */
    IShareButton.prototype.failIcon;
    /** @type {?|undefined} */
    IShareButton.prototype.share;
    /** @type {?|undefined} */
    IShareButton.prototype.count;
}
/**
 * Share button directive ref interface
 * This ref to be used in the share operators
 * @record
 */
export function ShareButtonRef() { }
function ShareButtonRef_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ShareButtonRef.prototype.prop;
    /** @type {?|undefined} */
    ShareButtonRef.prototype.renderer;
    /** @type {?|undefined} */
    ShareButtonRef.prototype.cd;
    /** @type {?|undefined} */
    ShareButtonRef.prototype.el;
    /** @type {?|undefined} */
    ShareButtonRef.prototype.os;
    /** @type {?|undefined} */
    ShareButtonRef.prototype.temp;
    /** @type {?} */
    ShareButtonRef.prototype.metaTags;
}
